src/ripple/rpc/impl/ShardArchiveHandler.cpp

```diff
@@ -61,18 +64,45 @@ ShardArchiveHandler::~ShardArchiveHandler()
 }
 
 bool
-ShardArchiveHandler::init()
+ShardArchiveHandler::add(std::uint32_t shardIndex, parsedURL&& url)
+{
+    std::lock_guard<std::mutex> l(m_);
```

Comment:
nit: the "l" variable is hard to read, can we use `lock` instead?  (applies to the whole PR)